hibernate-core/src/main/java/org/hibernate/type/descriptor/java/DurationJavaDescriptor.java

```diff
@@ -30,15 +44,22 @@ public String toString(Duration value) {
 		if ( value == null ) {
 			return null;
 		}
-		return String.valueOf( value.toNanos() );
+		return new StringBuilder(22)
+				.append( value.getSeconds() )
+				.append( value.getNano() )
+				.toString();
 	}
 
 	@Override
 	public Duration fromString(String string) {
 		if ( string == null ) {
 			return null;
 		}
-		return Duration.ofNanos( Long.parseLong( string ) );
+		int cutoff = string.length() - 9;
+		return Duration.ofSeconds(
+				Long.parseLong( string.substring( 0, cutoff ) ),
+				Long.parseLong( string.substring( cutoff ) )
```

Comment:
The above code logic seems based on the assumption that nano part's length is of 9, which is not necessarily true.

Reply:
That bit looks perfectly fine to me.